### PR TITLE
Update category model

### LINKED ISSUE #57: Add category tests
We need to have tests for the category model.

### PR DESCRIPTION
**Pull request by Ofekmn**
adds verification and the ability to update.

resolves #57

### CODE REVIEW

**Review by talko611 — COMMENTED** [review]
LGTM, please add update method tests

**Review by rotem2022 — APPROVED** [review]

File: category/tests.py [hunk 1]
```diff
@@ -1,3 +1,31 @@
...
+        new_category_name = "category2"
+        category1.update(new_category_name)
+        category1.save()
+        updated_category = Category.objects.get(id=category1.id)
+        assert updated_category.name == new_category_name
+
+    def test_save(self):
+        category = Category(name=CATEGORY_NAME)
+        category.save()
+        assert category in Category.objects.all()
```
Comment (kobihk) [inline category/tests.py:26]: if you add unique=true as I suggested in the model file I think you can remove these lines
```suggestion

```

File: category/tests.py [hunk 2]
```diff
@@ -1,3 +1,31 @@
...
+
+    def test_category_with_same_name(self):
+        Category.objects.create(name=CATEGORY_NAME)
```
Comment (kobihk) [inline category/tests.py:29]: you can use the fixture category1 like:
```suggestion
    def test_category_with_same_name(self, category1):
```

File: category/tests.py [hunk 3]
```diff
@@ -1,3 +1,31 @@
...
+        with pytest.raises(IntegrityError):
+            Category.objects.create(name=CATEGORY_NAME)
```
Comment (kobihk) [inline category/tests.py:31]: please add validation tests like:
 * long name
 * empty name
 * special characters in the name

**Review by kobihk — CHANGES_REQUESTED** [review]

File: category/models.py [hunk 4]
```diff
@@ -1,8 +1,17 @@
...
+
+
+def alpha(value):
+    if not str(value).isalpha():
+        raise ValidationError("Alphabetic characters only")
 
 
 class Category(models.Model):
-    name = models.CharField(max_length=20)
+    name = models.CharField(max_length=20, unique=True, validators=[alpha])
```
Comment (kobihk) [inline category/models.py:11]: Do we really want only alpha?
names like:
 * test category(with space)
 * test_category(with underscore)
will be invalid

**Review by kobihk — APPROVED** [review]

**Review by kobihk — DISMISSED** [review]

**Review by SaarOrus — APPROVED** [review]
LGTM

**Review by kobihk — APPROVED** [review]

**Review by rotem2022 — APPROVED** [review]

**Review by Sendy459 — APPROVED** [review]

**Review by talko611 — APPROVED** [review]

**Review by ygalblum — APPROVED** [review]

**Pull request merged**